src/daemon/daemon.cpp

```diff
@@ -3019,3 +3022,30 @@ void mp::Daemon::finish_async_operation(const std::string& async_future_key)
     if (async_op_result.status_promise)
         async_op_result.status_promise->set_value(async_op_result.status);
 }
+
+void mp::Daemon::update_manifests_all(bool is_force_update_from_network)
+{
+    std::vector<std::future<void>> empty_futures;
+    empty_futures.reserve(config->image_hosts.size());
+    for (const auto& image_host : config->image_hosts)
+    {
+        empty_futures.emplace_back(std::async(std::launch::async, &VMImageHost::update_manifests, image_host.get(),
+                                              is_force_update_from_network));
+    }
+    for (auto& empty_future : empty_futures)
+    {
+        empty_future.get();
+    }
+}
+
+void mp::Daemon::wait_update_manifests_all_and_optionally_applied_force(bool force_manifest_network_download)
```

Comment:
That is a mouthful for a function name :wink: 

But, `const bool`?

Reply:
Yep, but I also did not have a better name in mind which can describe what he does. Anyway, it is just a pre-process step that needs to be applied before retrieving images in the daemon::find function.

Reply:
The name is ok, just sayin'.  But the `const bool` stands. :slightly_smiling_face: